Docs/Specifics/Vision.md

```diff
@@ -2,64 +2,113 @@
+This trades the amount of information in each tag for quick, easy, and robust detection.
+There are several different "families" of tags depending on how many bits a tag can represent.
+In 2023, the FRC game Charged Up used 16h5, while in 2024 (and presumably onwards) 36h11 was used instead.
+In each frame in the camera stream, we are looking for these tags.
+Through a series of image processing algorithms (removing colors, decimation, identifying regions of relative lightness/darkness), we're able to identify both *which* tag it is (a numerical ID corresponding to the unique pattern of the tag) and *where* the corners of the tag are in the frame.
+
+In the context of FRC, because AprilTags are placed in various known locations around the field, we can use them in order for the robot to know where it is on the field.
+The simplest application of this is single tag detection and 2D alignment.
+For example, if we know that when a certain tag is in the middle of the camera's field of vision, our robot's shooter is lined up correctly with the target, we can write code that instructs the shooter to release a game piece once that tag has been centered, the drivetrain to adjust so that the tag is centered, etc.
+
+### Pose Estimation
+We can also use AprilTags in a more sophisticated way to perform pose estimation.
+Once a tag and its corner locations have been detected, we can compare that to the known pose of that tag on the field and calculate an estimate of the distance, angle, etc. that the camera is from that tag.
+We know where the camera is relative to the rest of the robot, so we can then estimate where the robot is on the field.
+We'll also combine this estimate with the estimated pose returned by the wheels and gyro of the robot, which is known as odometry.
+This is useful because with a reliable pose estimation system, we can automate more of the scoring process.
+For example, in 2024 we were able to develop an accurate auto aim system based on the robot's estimated pose.
+
+<img src="../../Assets/Pose.PNG" alt="An example of pose estimation" width="600"/>
+
+Pose estimation is more accurate when there are multiple tags in frame.
+Each tag generates its own estimate, which can then be combined and used or discarded if one is inaccurate.
+
+How would an estimate be incorrect?
+One cause of this is pose ambiguity.
+
+![Demonstration of pose ambiguity](/Assets/Ambiguity.jfif)
+
+Humans are typically able to rely on things like depth perception and lighting to determine the difference between something like these two boxes, but it is more difficult for computers.
+If we also have pose estimates from other tags that are less ambiguous, we can throw out those (likely) incorrect estimates if they're significantly off.
+Having multiple cameras also helps with this, as another camera at a different angle could generate a more accurate pose with the same tag.
+Comparing a vision pose against odometry can also rule out incorrect poses.
+
```

Comment:
i would add a short paragraph explaining different disambiguation techniques and ambiguity, such as lowest ambiguity, closest to floor, and PNP.